pkg/controller/dataimportcron-controller.go

```diff
@@ -702,6 +714,9 @@ func (r *DataImportCronReconciler) handleSnapshot(ctx context.Context, dataImpor
 		},
 	}
 	r.setDataImportCronResourceLabels(dataImportCron, desiredSnapshot)
+	if pvc.Spec.VolumeMode != nil {
+		cc.AddAnnotation(desiredSnapshot, cc.AnnSourceVolumeMode, string(*pvc.Spec.VolumeMode))
```

Comment:
If Im not missing anything the annotation will only be added in case we dont find the snapshot in line 722 and then we will create it.. if so we should do this change inside the `if`

Reply:
this path sets the ann. on newly created snapshots, existing ones will be handled at 360

Reply:
yeah I meant that unless you get inside the if in line 722, this will not actually do anything, so it should be under it and not before it

Reply:
added